ecosystem/indexer-grpc/indexer-transaction-generator/src/transaction_importer.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::TransactionImporterPerNetworkConfig;
+use anyhow::Context;
+use aptos_indexer_grpc_utils::create_data_service_grpc_client;
+use aptos_protos::indexer::v1::GetTransactionsRequest;
+use std::{path::Path, time::Duration};
+
+/// GRPC request metadata key for the token ID.
+const GRPC_API_GATEWAY_API_KEY_HEADER: &str = "authorization";
+const GRPC_REQUEST_NAME_HEADER: &str = "x-request-name";
+const GRPC_REQUEST_NAME_VALUE: &str = "testing-framework";
+const TRANSACTION_STREAM_TIMEOUT_IN_SECS: u64 = 60;
+
+impl TransactionImporterPerNetworkConfig {
+    pub async fn run(&self, output_path: &Path) -> anyhow::Result<()> {
+        let mut client = create_data_service_grpc_client(
+            self.transaction_stream_endpoint.clone(),
+            Some(Duration::from_secs(TRANSACTION_STREAM_TIMEOUT_IN_SECS)),
+        )
+        .await?;
+
+        for (version, output_file) in &self.versions_to_import {
+            let mut request = tonic::Request::new(GetTransactionsRequest {
+                starting_version: Some(*version),
+                transactions_count: Some(1),
+                ..GetTransactionsRequest::default()
+            });
+            request.metadata_mut().insert(
+                GRPC_REQUEST_NAME_HEADER,
+                GRPC_REQUEST_NAME_VALUE.parse().unwrap(),
+            );
+            if let Some(api_key) = &self.api_key {
+                request.metadata_mut().insert(
+                    GRPC_API_GATEWAY_API_KEY_HEADER,
+                    format!("Bearer {}", api_key.clone()).parse().unwrap(),
+                );
+            }
+            let mut stream = client.get_transactions(request).await?.into_inner();
+            while let Some(resp) = stream.message().await.context(format!(
+                "[Transaction Importer] Stream ended unexpected for endpoint {:?}",
+                self.transaction_stream_endpoint
+            ))? {
+                let transaction = resp.transactions.first().context(format!(
+                    "[Transaction Importer] Transaction at version {} is not in response.",
+                    version
+                ))?;
+                let json_string = serde_json::to_string_pretty(transaction).context(
+                    format!("[Transaction Importer] Transaction at version {} failed to serialized to json string.", version))?;
+                let output_path = output_path.join(output_file).with_extension("json");
```

Comment:
Do we want to be more strict about how the naming of the output files should be? For example, I think we should always include the network + transaction version in the name so that processors downstream can easily lookup the transactions in explorer and debug further. Here, we can make sure to prefix it with network + version.

Reply:
These transactions are controlled by the config and with the file name alone you can identify the source easily in the config. 

To use these files,
```
user testing_transactions::{IMPORTED_FUND_TRANSACTION, IMPORTED_BURN_TOKENS, GENERATED_KEYLESS_THING}
```

Or we want something like this
```
user testing_transactions::{IMPORTED_TESTNET_FUND_TRANSACTION_123123, IMPORTED_MAINNET_BURN_TOKENS_58473, GENERATED_KEYLESS_THING}
```

Then the question is whether we want downstream users to debug the potential issue during transaction generation.  It's doable but it may spam

Reply:
debuggability-wise, it's one more hop to look up in config; but readability may degrade.

Reply:
I prefer to have more explicit naming like 2nd one you provided. <imported | scripted>.<Testnet | Mainnet >. <transaction name(if scripted) | version (if imported)>.

Reply:
We could probably discuss this with the broader team when we finalize the process for m3.

Reply:
Prefer more explicit naming, like including at least the network + version for imported transaction. 

If we're going to be importing these in the processors repo, it's a hassle to lookup the config in internal-ops. It's much easier to get the version and lookup in explorer or grpcurl if needed.

Reply:
sounds good. let's update this in the next PR.